### DISCUSSION TITLE
How to copy all current tabs' URL into a single file using userscript?

### DISCUSSION THREAD

**Discussion by amalgame21** [qutebrowser/qutebrowser#7472] [Q&A] [answered]
Now I have a userscript to yank current tab's URL into a file:
```
#!/usr/bin/bash

echo $QUTE_URL >> ~/URL.txt
```

However, if I have 100 tabs, and I want to copy all the tabs' URL into the `URL.txt` file, I have to switch to the next tab, and do the userscript 100 times.

How to create a userscript that can read current total number of tabs and copy all those tabs' URL into a file?

I see a option for `$QUTE_TAB_INDEX` which maybe relevant to this purpose, but I do not know how to use it to make qutebrowser to switch to another tabs.

Thanks!

**Comment by The-Compiler** [comment, marked as answer]
That's basically what `:session-save` does. You can also pass a file name to it as well. If you need to get only the URLs from that, you can parse the resulting YAML file. What's your use-case/goal?

**Reply by amalgame21** [reply]
I opened a lot of jpg links and see which one I would like to download and keep that tab, and I want to do a mass download of them by `wget` later on.
I think the YAML file of `:session-save` combined with `grep "url:" default.yaml` is a better solution for that, thanks for your good advice!
